.github/workflows/release.yml

```
@@ -2,86 +2,44 @@ name: Release
 
 on:
   push:
+    branches: [main]
     tags:
       - "v*"
+  pull_request:
+    branches: [main]
 
 jobs:
   build:
     strategy:
+      fail-fast: false
       matrix:
         include:
           - target: x86_64-unknown-linux-gnu
             os: ubuntu-latest
+          - target: aarch64-unknown-linux-gnu
+            os: ubuntu-latest
           - target: x86_64-pc-windows-gnu
             os: ubuntu-latest
             extension: .exe
-          - target: x86_64-apple-darwin
+          # - target: x86_64-apple-darwin
+          #   os: ubuntu-latest
+          - target: aarch64-apple-darwin
             os: macos-latest
 
     runs-on: ${{ matrix.os }}
 
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions-rs/toolchain@v1
-        with:
-          toolchain: stable
-      - uses: actions-rs/cargo@v1
-        with:
-          command: build
-          args: --release --target=${{ matrix.target }}
-          use-cross: true
-      - uses: actions/upload-artifact@v1
-        with:
-          name: build-${{ matrix.target }}
-          path: target/${{ matrix.target }}/release/split-test${{ matrix.extension }}
-
-  create-release:
-    needs: [build]
-    runs-on: ubuntu-latest
-    steps:
-      - id: create-release
-        uses: actions/create-release@v1
+      - uses: actions/checkout@v4
+      - run: rustup update stable
+      - run: rustup default stable
```

According to the error message

https://github.com/int128/split-test/actions/runs/8845858159/job/24290599577

```
error[E0463]: can't find crate for `core`
  |
  = note: the `x86_64-apple-darwin` target may not be installed
  = help: consider downloading the target with `rustup target add x86_64-apple-darwin`
```

perhaps this patch helps?

```suggestion
      - run: rustup default stable
      - run: rustup target add ${{ matrix.target }}
```

---

@int128

---

@int128 Do you still have interests on this? At least I'd like to fix the CI if it has any trouble

---

Sorry, I forgot everything about this pull request. If needed, feel free to pick my changes.